src/main/java/com/birdie/kafka/connect/smt/Outbox.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    public static final ConfigDef CONFIG_DEF = new ConfigDef()
+        .define(Outbox.ConfigName.TOPIC, ConfigDef.Type.STRING, ConfigDef.Importance.MEDIUM, "The name of the topic to send messages to.")
+        .define(Outbox.ConfigName.PARTITION_SETTING, ConfigDef.Type.STRING, "partition-number", ConfigDef.Importance.MEDIUM, "Set to \"partition-number\" (default) to use the record's partition_number value, or set to \"partition-key\" to generate partition number using record's partition_key value")
+        .define(Outbox.ConfigName.NUMBER_OF_PARTITION_IN_TOPIC, ConfigDef.Type.INT, 0, ConfigDef.Importance.MEDIUM, "Number of partitions on the target topic")
+        .define(Outbox.ConfigName.DEBUG_LOG_LEVEL, ConfigDef.Type.STRING, "trace", ConfigDef.Importance.LOW, "Log level for debugging purposes")
+    ;
+
+    private String targetTopic;
+    private PartitionSetting partitionSetting;
+    private Integer numberOfPartitionsInTargetTopic;
+    private DebugLogLevel debugLogLevel;
+
+    public enum PartitionSetting {
+        PARTITION_KEY, PARTITION_NUMBER
+    }
+
+    @java.lang.Override
+    public void configure(Map<String, ?> props) {
+        final SimpleConfig config = new SimpleConfig(CONFIG_DEF, props);
+
+        targetTopic = config.getString(ConfigName.TOPIC);
+        numberOfPartitionsInTargetTopic = config.getInt(ConfigName.NUMBER_OF_PARTITION_IN_TOPIC);
+
+        String partitionSettingString = config.getString(ConfigName.PARTITION_SETTING);
+        try {
+            partitionSetting = PartitionSetting.valueOf(partitionSettingString.toUpperCase().replace('-', '_'));
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid partition setting provided: " + partitionSettingString, e);
+        }
+        if (partitionSetting == PartitionSetting.PARTITION_KEY && numberOfPartitionsInTargetTopic == 0) {
+            throw new IllegalArgumentException("num-target-partitions is zero/null, when auto-partitioning is set to true");
+        }
+        
+        String debugLogLevelString = config.getString(ConfigName.DEBUG_LOG_LEVEL);
+        try {
+            debugLogLevel = DebugLogLevel.valueOf(debugLogLevelString.toUpperCase());
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid log level provided: " + debugLogLevelString, e);
+        }
+    }
+
+    @java.lang.Override
+    public SourceRecord apply(SourceRecord sourceRecord) {
+        DEBUG("Received source record: {}", sourceRecord);
+
+        if (sourceRecord.value() == null) {
+            DEBUG("Dropping debezium-generated tombstones with null partition_key: {}", sourceRecord);
+            return null;
+        }
+
+        Schema valueSchema = sourceRecord.valueSchema();
+        if (valueSchema.name().equals("io.debezium.connector.common.Heartbeat")) {
+            DEBUG("Ignoring debezium-connector heartbeat: {}", sourceRecord);
+            return sourceRecord;
+        }
```

Comment:
tbh I wasn't 100% sure from the documentation, if these heartbeats could be dropped, or they are necessary.

Their main purpose is to prevent the offset on the replication slot from going stale and nuking the DB storage (replication slot lag), but I don't know if it's enough to get a heartbeat up to the SMT point, or it needs to be passed to the target topic.

For now am allowing the record to pass through unmodified, which means it goes to the topic `__debezium-heartbeat.matching`

Reply:
Hearbeats could be dropped if you remove this [line](https://github.com/birdiecare/matching/pull/73/files#diff-2545cc7799b1df233c28343e71a6b0ff4d1e87e6aa9608d6b2844f6351d547efR68), but I think that it's better to maintain them for reliability/performance. You can check the [documentation](https://debezium.io/documentation/reference/connectors/postgresql.html#postgresql-property-heartbeat-interval-ms).